tests/src/main/java/net/neoforged/neoforge/debug/registry/BiomeModifierSyncTest.java

```diff
@@ -0,0 +1,116 @@
+import net.neoforged.testframework.TestFramework;
+import net.neoforged.testframework.TestListener;
+import net.neoforged.testframework.annotation.ForEachTest;
+import net.neoforged.testframework.annotation.OnInit;
+import net.neoforged.testframework.annotation.TestHolder;
+import net.neoforged.testframework.annotation.WithListener;
+import net.neoforged.testframework.gametest.EmptyTemplate;
+import net.neoforged.testframework.registration.RegistrationHelper;
+import org.jetbrains.annotations.Nullable;
+
+@ForEachTest(groups = BiomeModifierSyncTest.GROUP)
+public class BiomeModifierSyncTest {
+    public static final String GROUP = "registry";
+    public static final RegistrationHelper HELPER = RegistrationHelper.create("neotests_biome_modifier_sync_test");
+
+    private static final int MODIFIED_WATER_COLOR = 0x0000FF;
+
+    private static final DeferredRegister<MapCodec<? extends BiomeModifier>> BIOME_MODIFIER_SERIALIZERS = HELPER.registrar(NeoForgeRegistries.Keys.BIOME_MODIFIER_SERIALIZERS);
+    private static final DeferredHolder<MapCodec<? extends BiomeModifier>, MapCodec<TestModifier>> MODIFY_BIOME_EFFECTS = BIOME_MODIFIER_SERIALIZERS.register("modify_biome_effects", () -> RecordCodecBuilder.mapCodec(builder -> builder.group(
+            Biome.LIST_CODEC.fieldOf("biomes").forGetter(TestModifier::biomes),
+            Codec.INT.fieldOf("water_color").forGetter(TestModifier::waterColor)).apply(builder, TestModifier::new)));
+
+    public record TestModifier(HolderSet<Biome> biomes, int waterColor) implements BiomeModifier {
+        @Override
+        public void modify(Holder<Biome> biome, Phase phase, ModifiableBiomeInfo.BiomeInfo.Builder builder) {
+            if (phase == Phase.MODIFY && this.biomes.contains(biome)) {
+                builder.getSpecialEffects().waterColor(this.waterColor);
+            }
+        }
+
+        @Override
+        public MapCodec<? extends BiomeModifier> codec() {
+            return MODIFY_BIOME_EFFECTS.get();
+        }
+    }
+
+    @OnInit
+    static void init(final TestFramework framework) {
+        HELPER.register(framework.modEventBus(), framework.container());
+        ResourceKey<BiomeModifier> modifyTaigaModifier = ResourceKey.create(NeoForgeRegistries.Keys.BIOME_MODIFIERS, ResourceLocation.fromNamespaceAndPath(HELPER.modId(), "modify_taiga"));
+        HELPER.addProvider(event -> new DatapackBuiltinEntriesProvider(
+                event.getGenerator().getPackOutput(),
+                event.getLookupProvider(),
+                new RegistrySetBuilder().add(NeoForgeRegistries.Keys.BIOME_MODIFIERS, context -> {
+                    var taigaTag = context.lookup(Registries.BIOME).getOrThrow(BiomeTags.IS_TAIGA);
+                    context.register(modifyTaigaModifier, new TestModifier(
+                            taigaTag,
+                            MODIFIED_WATER_COLOR));
+                }),
+                Set.of(HELPER.modId())));
+    }
+
+    @GameTest
```

Comment:
Why is this marked as a gametest if it's client only?

Reply:
Probably because I'm not entirely clear on how your test framework works. If I want it to show up in the client test list, does it not have to be a gametest?

Reply:
Seems to work fine without it, so, cool!